cli/src/tasks/migrate.ts

```diff
@@ -0,0 +1,590 @@
+  // classpath 'com.android.tools.build:gradle:7.2.1'
+  // classpath 'com.google.gms:google-services:4.3.10'
+  const txt = readFile(filename);
+  if (!txt) {
+    return;
+  }
+  const neededDeps: { [key: string]: string } = {
+    'com.android.tools.build:gradle': '7.2.1',
+    'com.google.gms:google-services': '4.3.10',
+  };
+  let replaced = txt;
+
+  for (const dep of Object.keys(neededDeps)) {
+    if (!replaced.includes(`classpath '${dep}`)) {
+      replaced = txt.replace(
+        'dependencies {',
+        `dependencies {\n        classpath '${dep}:${neededDeps[dep]}'`,
+      );
+    } else {
+      // Update
+      replaced = setAllStringIn(
+        replaced,
+        `classpath '${dep}:`,
+        `'`,
+        neededDeps[dep],
+      );
+      logger.info(`Migrated build.gradle set ${dep} = ${neededDeps[dep]}.`);
+    }
+  }
+
+  // Replace jcenter()
+  const lines = replaced.split('\n');
+  let inRepositories = false;
+  let hasMavenCentral = false;
+  let final = '';
+  for (const line of lines) {
+    if (line.includes('repositories {')) {
+      inRepositories = true;
+      hasMavenCentral = false;
+    } else if (line.trim() == '}') {
+      // Make sure we have mavenCentral()
+      if (inRepositories && !hasMavenCentral) {
+        final += '        mavenCentral()\n';
+        logger.info(`Migrated build.gradle added mavenCentral().`);
+      }
+      inRepositories = false;
+    }
+    if (inRepositories && line.trim() === 'mavenCentral()') {
+      hasMavenCentral = true;
+    }
+    if (inRepositories && line.trim() === 'jcenter()') {
+      // skip jCentral()
+      logger.info(`Migrated build.gradle removed jcenter().`);
+    } else {
+      final += line + '\n';
+    }
+  }
+
+  if (txt !== final) {
+    writeFileSync(filename, final, 'utf-8');
+    return;
+  }
+}
+
+function readFile(filename: string): string | undefined {
+  try {
+    if (!existsSync(filename)) {
+      logger.error(`Unable to find ${filename}. Try updating it manually`);
+      return;
+    }
+    return readFileSync(filename, 'utf-8');
+  } catch (err) {
+    logger.error(
+      `Unable to read ${filename}. Verify it is not already open. ${err}`,
+    );
+  }
+}
+
+function updateAppBuildGradle(filename: string) {
+  const txt = readFile(filename);
+  if (!txt) {
+    return;
+  }
+  if (txt.includes('androidx.coordinatorlayout:coordinatorlayout:')) {
+    return;
+  }
+
+  const replaced = txt.replace(
+    'dependencies {',
+    'dependencies {\n    implementation "androidx.coordinatorlayout:coordinatorlayout:$androidxCoordinatorLayoutVersion"',
+  );
+  // const lines = txt.split('\n');
+  writeFileSync(filename, replaced, 'utf-8');
+  logger.info(
+    `Migrated ${filename} by adding androidx.coordinatorlayout dependency.`,
+  );
+}
+
+function updateGradleWrapper(filename: string) {
+  const txt = readFile(filename);
+  if (!txt) {
+    return;
+  }
+  let replaced = txt;
+  if (replaced.includes('gradle-7.0-all.zip')) {
```

Comment:
Since Android Studio prompts users to update gradle from time to time, it's possible that they are in a newer version than 7.0, so I think this should just replace whatever is on the url with 7.4.2